### PR TITLE
feat: highlight text

### PR DESCRIPTION
**Pull request by anhle10051996**
## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot changes introduced" width="250" />
OR
Video link

### CODE REVIEW

File: example/src/stories/HighlightText.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export default {
+  title: 'components/HighlightText',
+  component: HighLightText,
+} as ComponentMeta<typeof HighLightText>
+
+export const Basic: ComponentStory<typeof HighLightText> = args => <HighLightText {...args} />
+
+Basic.args = {
+  textToHighlight: 'Hello STS Tea123123m!',
```
Comment (huydosgtech) [inline example/src/stories/HighlightText.stories.tsx:14]: Doest this text "Tea123123m" have meaning?

File: src/__tests__/HighlightText.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import {render} from '@testing-library/react-native'
+import HighlightText from '../components/HighlightText/HighlightText'
+import type {ReactTestInstance} from 'react-test-renderer'
+import {StyleSheet} from 'react-native'
+
+describe('HighlightText', () => {
+  const textToHighlight = 'Lorem ipsum dolor sit amet consectetur adipiscing elit'
```
Comment (huydosgtech) [inline src/__tests__/HighlightText.test.tsx:8]: can we use english for easy to read instead another language?

File: src/components/HighlightText/HighlightText.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  sanitize,
+  autoEscape = false,
+  textWrapperStyle,
+  normalTextStyle,
+  highlightTextStyle,
+  ...props
+}) => {
+  const chunks = findAll({textToHighlight, searchWords, sanitize, autoEscape})
+  return (
+    <TextWrapper style={textWrapperStyle} testID="container" {...props}>
```
Comment (huydosgtech) [inline src/components/HighlightText/HighlightText.tsx:61]: can we make testID is prop instead hardcode? 
exp: testID=`${testID}-container`

File: example/src/stories/HighlightText.stories.tsx (hunk 1 above)
Comment (anhle10051996) [reply example/src/stories/HighlightText.stories.tsx:14]: No meaning. Just for testing the text to highlight

File: src/components/HighlightText/HighlightText.tsx (hunk 3 above)
Comment (anhle10051996) [reply src/components/HighlightText/HighlightText.tsx:61]: i think it's not necessary because the testID only using for jest test

File: src/__tests__/HighlightText.test.tsx (hunk 2 above)
Comment (hangnguyensaigontech) [reply src/__tests__/HighlightText.test.tsx:8]: It's just dummy text for testing

**Review by baonguyenhsts — APPROVED** [review]

**Review by hangnguyensaigontech — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Review by vunguyendhSTS — APPROVED** [review]
